test/solver/gmres_kernels.cpp

```diff
@@ -305,11 +313,35 @@ TEST_F(Gmres, GmresApplyOneRHSIsEquivalentToRef)
     auto exec_solver = exec_gmres_factory->generate(d_mtx);
     auto b = gen_mtx(m, n);
     auto x = gen_mtx(m, n);
-    auto d_b = gko::clone(exec, b);
-    auto d_x = gko::clone(exec, x);
+    auto d_b = Mtx::create(exec);
+    auto d_x = Mtx::create(exec);
+    d_b->copy_from(b.get());
+    d_x->copy_from(x.get());
+
+    ref_solver->apply(b.get(), x.get());
+    exec_solver->apply(d_b.get(), d_x.get());
+
+    GKO_ASSERT_MTX_NEAR(d_b, b, 0);
+    GKO_ASSERT_MTX_NEAR(d_x, x, r<value_type>::value * 1e2);
+}
+
+
+TEST_F(Gmres, GmresApplyMultipleRHSIsEquivalentToRef)
+{
+    int m = 123;
+    int n = 5;
+    auto ref_solver = ref_gmres_factory->generate(mtx);
+    auto exec_solver = exec_gmres_factory->generate(d_mtx);
+    auto b = gen_mtx(m, n);
+    auto x = gen_mtx(m, n);
+    auto d_b = Mtx::create(exec);
+    auto d_x = Mtx::create(exec);
+    d_b->copy_from(b.get());
+    d_x->copy_from(x.get());
 
     ref_solver->apply(b.get(), x.get());
     exec_solver->apply(d_b.get(), d_x.get());
 
-    GKO_ASSERT_MTX_NEAR(d_x, x, r<value_type>::value * 100);
+    GKO_ASSERT_MTX_NEAR(d_b, b, 0);
+    GKO_ASSERT_MTX_NEAR(d_x, x, r<value_type>::value * 1e3);
```

Comment:
do you know which case needs larger criterion?

Reply:
No, I just left the values in there it already had. I am also not sure why `d_b` and `b` is checked as it should be a direct copy.